### PR TITLE
test: fix flaky test due to static state

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

The `TlsTest` tended to fail locally when running other tests in parallel. Investigation showed this is due to shared state in `ServerKeyStore`: the path to the key store.  The test needed to write a bad cert and then assert requests failed, but if another test ran it would override the bad cert with a good one.

### CODE REVIEW

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TlsTest.java [hunk 1]
```diff
@@ -35,22 +35,20 @@
 import java.util.concurrent.TimeUnit;
 import org.apache.kafka.common.config.SslConfigs;
 import org.junit.Test;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
-public class TlsTest extends ApiTest {
+public class TlsTest extends BaseApiTest {
```
Comment (vcrfxia) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TlsTest.java:39]: Hey @big-andy-coates , can we undo this change? It's intentional that TlsTest extends ApiTest so we have test coverage for issuing requests to a TLS-enabled server.

**Comment by vcrfxia** [comment]
> The TlsTest tended to fail locally when running other tests in parallel.

For my edification, when are tests run in parallel? I thought our tests were always run sequentially...?

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TlsTest.java (hunk 1 above)
Comment (big-andy-coates) [reply ksqldb-rest-app/src/test/java/io/confluent/ksql/api/TlsTest.java:39]: Done in https://github.com/confluentinc/ksql/pull/6144
